src/System.Memory/tests/ReadOnlyBuffer/ReadOnlySequenceTests.Common.cs

```diff
@@ -303,120 +372,411 @@ public void EnumerableStopsAtEndWhenEndIsFirstByteOfEmpty()
+            Assert.Throws<ArgumentOutOfRangeException>("offset", () => buffer.GetPosition(1, buffer.End));
         }
 
         [Fact]
-        public void Empty_ReturnsLengthZeroBuffer()
+        public void Empty_GetPositionNegative()
         {
-            var buffer = ReadOnlySequence<byte>.Empty;
+            ReadOnlySequence<byte> buffer = ReadOnlySequence<byte>.Empty;
+            Assert.Throws<ArgumentOutOfRangeException>("offset", () => buffer.GetPosition(-1));
+            Assert.Throws<ArgumentOutOfRangeException>("offset", () => buffer.GetPosition(-1, buffer.Start));
+            Assert.Throws<ArgumentOutOfRangeException>("offset", () => buffer.GetPosition(-1, buffer.End));
+        }
+
+        [Fact]
+        public void Empty_TryGet()
+        {
+            ReadOnlySequence<byte> buffer = ReadOnlySequence<byte>.Empty;
+            ReadOnlyMemory<byte> memory;
+
+            SequencePosition c1 = buffer.Start;
+            Assert.True(buffer.TryGet(ref c1, out memory, false));
+            Assert.Equal(buffer.Start, c1);
+            Assert.True(memory.IsEmpty);
+
+            Assert.True(buffer.TryGet(ref c1, out memory, true));
+            Assert.Equal(null, c1.GetObject());
+            Assert.True(memory.IsEmpty);
+
+            Assert.False(buffer.TryGet(ref c1, out memory, false));
+            Assert.Equal(null, c1.GetObject());
+            Assert.True(memory.IsEmpty);
+            
+            Assert.False(buffer.TryGet(ref c1, out memory, true));
+            Assert.Equal(null, c1.GetObject());
+            Assert.True(memory.IsEmpty);
+
+            Assert.False(buffer.TryGet(ref c1, out memory, false));
+            Assert.Equal(null, c1.GetObject());
+            Assert.True(memory.IsEmpty);
+
+            Assert.False(buffer.TryGet(ref c1, out memory, true));
+            Assert.Equal(null, c1.GetObject());
+            Assert.True(memory.IsEmpty);
+
+            c1 = buffer.End;
+            Assert.True(buffer.TryGet(ref c1, out memory, false));
+            Assert.Equal(buffer.End, c1);
+            Assert.True(memory.IsEmpty);
+
+            Assert.True(buffer.TryGet(ref c1, out memory, true));
+            Assert.Equal(null, c1.GetObject());
+            Assert.True(memory.IsEmpty);
+
+            Assert.False(buffer.TryGet(ref c1, out memory, false));
+            Assert.Equal(null, c1.GetObject());
+            Assert.True(memory.IsEmpty);
+            
+            Assert.False(buffer.TryGet(ref c1, out memory, true));
+            Assert.Equal(null, c1.GetObject());
+            Assert.True(memory.IsEmpty);
+
+            Assert.False(buffer.TryGet(ref c1, out memory, false));
+            Assert.Equal(null, c1.GetObject());
+            Assert.True(memory.IsEmpty);
+
+            Assert.False(buffer.TryGet(ref c1, out memory, true));
+            Assert.Equal(null, c1.GetObject());
+            Assert.True(memory.IsEmpty);
+        }
+        
+        #endregion
+
+        #region default
```

Comment:
nit: Consider splitting these tests into separate files for each of the APIs it is testing (rather than using regions).

Similar to ReadOnlySequenceTests.TryGet.cs (or the [Span tests](https://github.com/dotnet/corefx/tree/f592e887e2349ed52af6a83070c42adb9d26408c/src/System.Memory/tests/Span)).